nima_core/cache.py

```
@@ -0,0 +1,254 @@
+                self._cache_hits += 1
+                continue
+
+            # Needs computation
+            to_compute.append(text)
+            to_compute_idxs.append(i)
+            results.append((i, None))
+
+        # Batch compute missing
+        if to_compute:
+            computed = embed_batch(to_compute)
+            for idx, text, emb in zip(to_compute_idxs, to_compute, computed):
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)